turbine/src/sigverify_shreds.rs

```diff
@@ -30,6 +37,9 @@ const DEDUPER_FALSE_POSITIVE_RATE: f64 = 0.001;
 const DEDUPER_NUM_BITS: u64 = 637_534_199; // 76MB
 const DEDUPER_RESET_CYCLE: Duration = Duration::from_secs(5 * 60);
 
+const CLUSTER_NODES_CACHE_NUM_EPOCH_CAP: usize = 8;
+const CLUSTER_NODES_CACHE_TTL: Duration = Duration::from_secs(5);
```

Comment:
I know we don't do it anywhere else, but it would be nice to have a comment explaining why we chose these numbers.

8 epochs worth of cluster nodes seems excessive, but I also don't really know why we would need to look back at historical epochs.

Similarly, the tradeoffs between shorter/longer time-to-live are not clear. I'm guessing fresher data vs. update overhead, but it's not clear exactly how out of date we're allowed to be

Reply:
Added some clarifying comments to the code and made some adjustments.

Technically we only need 2 epochs because near an epoch boundary you might receive shreds from the other side of the epoch boundary. I have been a bit paranoid here to pick a larger value just to cover unknown unknowns, like in case of restart after a long outage (or say with `warp-slot` or `warp-epoch`), or long network partition.

> 8 epochs worth of cluster nodes seems excessive, but I also don't really know why we would need to look back at historical epochs.

I reduced this to 5, and using `MAX_LEADER_SCHEDULE_STAKES` directly since it does not make sense to have capacity more than 5 when there is TTL expiration and only 5 epoch-stakes in bank.

> Similarly, the tradeoffs between shorter/longer time-to-live are not clear. I'm guessing fresher data vs. update overhead, but it's not clear exactly how out of date we're allowed to be

Shorter TTL will pull gossip more often to refresh gossip contact-infos. However here we only need pubkeys of staked nodes so gossip contact-infos are redundant. I increased this to 30 seconds.